src/panels/lovelace/common/compute-card-size.ts

```diff
@@ -1,5 +1,13 @@
 import { LovelaceCard } from "../types";
 
-export const computeCardSize = (card: LovelaceCard): number => {
-  return typeof card.getCardSize === "function" ? card.getCardSize() : 4;
+export const computeCardSize = async (card: LovelaceCard): Promise<number> => {
+  if (typeof card.getCardSize === "function") {
+    return card.getCardSize();
+  }
+  if (customElements.get(card.localName)) {
+    return 1;
+  }
+  return customElements.whenDefined(card.localName).then(() => {
+    return computeCardSize(card);
+  });
```

Comment:
Should we do `await` here to make the stack traces nicer ? 

```suggestion
  await customElements.whenDefined(card.localName)
  return computeCardSize(card);
```